binarytree/__init__.py

```diff
@@ -1031,6 +1064,57 @@ def size(self):
         """
         return _get_tree_properties(self)['size']
 
+    @property
+    def wtree(self):
+        """Set the input tree as a search space.
+        """
+        return self._root
+
+    @wtree.setter
+    def wtree(self, tree):
+        """The setter function for wtree setting.
+
+        :param tree: The start root node of the binary tree.
+        :rtype: binarytree.Node
+        """
+        self._root = tree
+
+    @property
+    def upper(self):
```

Comment:
Why not just call this `parent`?

Reply:
Because when you call `left` you mean go to the left node and `right` to the right one. I just want to be the same style with these properties. eg. You can call `up` or `upper` to go to the upper node.

Reply:
The most common convention is "parent". I've never seen the term "upper node" used in context of binary trees so it may confuse the users.